onnxruntime/core/providers/webnn/builders/impl/normalization_op_builder.cc

```diff
@@ -44,10 +47,20 @@ Status NormalizationOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder
   ORT_RETURN_IF_NOT(GetShape(*input_defs[1], scale_shape, logger), "Cannot get scale shape");
   const auto scale_size = scale_shape.size();
   ORT_RETURN_IF_NOT(scale_size >= 1 && scale_size <= rank, "The scale size should be less than or equal to input size.");
-
+  // Enlarge new shape with ones.
   if (scale_size < rank) {
-    // Enlarge new shape to input.rank, right aligned with leading ones
-    scale_shape.insert(scale_shape.begin(), rank - scale_size, 1);
+    if (op_type == "LayerNormalization") {
+      scale_shape.insert(scale_shape.begin(), rank - scale_size, 1);
+    } else if (op_type == "InstanceNormalization") {
+      scale_shape.insert(scale_shape.begin(), 1);
+      scale_shape.insert(scale_shape.end(), rank - scale_size - 1, 1);
+    } else if (op_type == "GroupNormalization") {
+      scale_shape.insert(scale_shape.begin(), 1);
+      scale_shape.insert(scale_shape.end(), 1);
```

Comment:
Ah, it's because of the reshape to 3D below, right? Then let's add a comment here:

```
      // The input will be reshaped to 3D later. So just insert ones before the channel and after.
      scale_shape.insert(scale_shape.begin(), 1);
      scale_shape.insert(scale_shape.end(), 1);
```